sbi/inference/snpe/snpe_base.py

```diff
@@ -176,16 +181,7 @@ def __call__(
                     mcmc_method=self._mcmc_method,
                     get_potential_function=PotentialFunctionProvider(),
                 )
-            self._handle_x_o_wrt_amortization(x_o, x_shape, num_rounds)
-
-            # Check for NaNs in simulations.
-            is_valid_x, num_nans, num_infs = handle_invalid_x(x, exclude_invalid_x)
-            warn_on_invalid_x(num_nans, num_infs, exclude_invalid_x)
-
-            # XXX Rename bank -> rounds/roundwise.
-            self._theta_bank.append(theta[is_valid_x])
-            self._x_bank.append(x[is_valid_x])
-            self._prior_masks.append(prior_mask[is_valid_x])
+                self._handle_x_o_wrt_amortization(x_o, x_shape, num_rounds)
```

Comment:
This was outside the if clause previously, has it been shifted on purpose?

Reply:
yes, this is on purpose. Otherwise the warning would be given out every round.